packages/kit/src/runtime/client/client.js

```diff
@@ -1959,6 +1954,20 @@ function reset_focus() {
 	}
 }
 
+/**
+ * @param {URL} url
+ * @param {Array<import('./types').BranchNode | undefined>} branch
+ */
+function normalize_url(url, branch) {
+	/** @type {import('types').TrailingSlash} */
+	let slash = 'never';
+	for (const node of branch) {
+		if (node?.slash !== undefined) slash = node.slash;
+	}
+	url.pathname = normalize_path(url.pathname, slash);
+	url.search = url.search; // turn `/?` into `/`
+}
```

Comment:
can we avoid mutating the URL here and instead return a new one? it's _probably_ safe, but i've lost many hours of my life to these sorts of sneaky mutations in the past...

Reply:
Oh strange, it looks like some existing behavior (test: `Routing › redirects from /routing/? to /routing `) actually depends on the mutation happening. Will look into it.

Reply:
Sorry for the delay. I've reverted the commit that makes this function immutable in case anyone wants to use this branch for a temporary fix — will continue looking into getting a per-branch config without running the load function.